### PR TITLE
test: add windows test that worker service restarts process

### PR DESCRIPTION
**Pull request by YutongLi291**
### What was the problem/requirement? (What/Why)

On windows, the worker agent creates a service when running `install-deadline-worker` (without the `--no-service` argument). One responsibility of this service is to restart the worker agent process if it exits (e.g. a program crash).

We require automated end-to-end testing to ensure this functionality works as expected.

### What was the solution? (How)
Add a test that verifies that when windows worker shuts down by a SIGKILL signal, the worker service restarts the process.

Also moved around the other test in this class, so that we do not have to create 2 workers in this class, rather than one.
### What is the impact of this change?
Better code quality for the code that is responsible for restarting the worker.
### How was this change tested?
```
source .e2e_windows_infra.sh
hatch run e2e-test
```
### Was this change documented?
No
### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: test/e2e/test_worker_status.py [hunk 1]
```diff
@@ -130,3 +103,92 @@ def check_worker_processes_exist() -> None:
...
+            ), f"DeadlineWorker service is in unexpected status {service_check_result.stdout}"
+
+        check_service_is_running()
+
+        # Check that the worker process is running
+
+        def check_worker_processes_exist() -> None:
+            process_check_result = class_worker.send_command("Get-Process pythonservice")
+
+            assert process_check_result.exit_code == 0, "Worker agent process is not running"
```
Comment (hindleym) [inline test/e2e/test_worker_status.py:155]: Should the quote for the error start with an `f`?

Comment (YutongLi291) [reply test/e2e/test_worker_status.py:155]: Nope, f indicates f string which can be populated with variables. We do not need a variable inside of the string in this case.

File: test/e2e/test_worker_status.py [hunk 2]
```diff
@@ -17,31 +17,6 @@
 
 @pytest.mark.parametrize("operating_system", [os.environ["OPERATING_SYSTEM"]], indirect=True)
 class TestWorkerStatus:
-    def test_worker_lifecycle_status_is_expected(
```
Comment (leongdl) [inline test/e2e/test_worker_status.py:20]: This seems not related to the PR?

File: test/e2e/test_worker_status.py [hunk 3]
```diff
@@ -130,3 +103,92 @@ def check_worker_processes_exist() -> None:
 
         # Check that there are worker processes running
         check_worker_processes_exist()
+
+    @pytest.mark.skipif(
+        os.environ["OPERATING_SYSTEM"] == "linux",
```
Comment (leongdl) [inline test/e2e/test_worker_status.py:108]: This would fail if we were compiling on a MAC?

Comment (hindleym) [reply test/e2e/test_worker_status.py:108]: We only have Linux and Windows worker agents.

File: test/e2e/test_worker_status.py (hunk 2 above)
Comment (hindleym) [reply test/e2e/test_worker_status.py:20]: Was noted with the PR that this is moved to reduce the number of workers needed.

**Review by hindleym — APPROVED** [review]

**Review by AlexTranAmz — APPROVED** [review]
